### PR TITLE
HDL-TH ingtegration support

### PR DESCRIPTION
**Pull request by kimpaller**

### CODE REVIEW

File: src/sdg/Gauntlet.groovy [hunk 1]
```diff
@@ -337,6 +346,11 @@ def stage_library(String stage_name) {
                             echo "Executing board recovery..."
                             nebula(nebula_cmd)
                         }catch(Exception ex){
+                            if(gauntEnv.netbox_allow_disable){
+                                def message = "Disable by ${env.JOB_NAME} ${env.BUILD_NUMBER}"
+                                def disable_command = 'netbox.disable-board --board-name=' + board + ' --failure --reason=' + '"' + message + '"' + ' --power-off'
```
Comment (tagoylo) [inline src/sdg/Gauntlet.groovy:351]: Disabling the board will occur when recovery failed, right?

File: src/sdg/Gauntlet.groovy [hunk 2]
```diff
@@ -1577,23 +1591,60 @@ def update_agents() {
...
-        def s = gauntEnv.required_hardware.size()
-        def b = gauntEnv.boards.size()
         def rh = gauntEnv.required_hardware
+        def ab = gauntEnv.boards
+        def s = rh.size()
+        def b = ab.size()
         def filtered_board_list = []
         def filtered_agent_list = []
+        def found_rh = []
+        def special_naming_cases = [
```
Comment (tagoylo) [inline src/sdg/Gauntlet.groovy:1601]: Why is there special naming cases? Where is this naming based from?

Comment (kimpaller) [reply src/sdg/Gauntlet.groovy:1601]: > Why is there special naming cases? Where is this naming based from?

This for cases where boot partition folder name (the board naming convention we use in TH) does not match with that found in info.txt (which I assume was derived from linux project naming)

File: src/sdg/Gauntlet.groovy (hunk 1 above)
Comment (kimpaller) [reply src/sdg/Gauntlet.groovy:351]: yes

File: src/sdg/Gauntlet.groovy [hunk 3]
```diff
@@ -265,6 +270,10 @@ def stage_library(String stage_name) {
                         set_elastic_field(board, 'kernel_started', 'False')
                         set_elastic_field(board, 'linux_prompt_reached', 'False')
                         is_nominal_exception = true
+                    }else if (ex.getMessage().contains('Checksum does not match')){
+                        set_elastic_field(board, 'uboot_reached', 'True')
+                        set_elastic_field(board, 'kernel_started', 'True')
+                        set_elastic_field(board, 'linux_prompt_reached', 'True')
```
Comment (jpineda3) [inline src/sdg/Gauntlet.groovy:276]: set_elastic_field calls in line 238 above nebula verify checksum call have already set the three fields to True.  Is this still necessary here?

Comment (kimpaller) [reply src/sdg/Gauntlet.groovy]: https://github.com/sdgtt/jenkins-shared-library/pull/81/commits/81d09121e6dc4638c08b402ed82f3c1cb4c5bed8

**Review by tagoylo — APPROVED** [review]

**Pull request merged**